File: server/bcephal-accounting-service/src/main/java/com/moriset/bcephal/accounting/domain/BookingBrowserData.java
/**
 * 
 */
package com.moriset.bcephal.accounting.domain;

import java.math.BigDecimal;
import java.util.Date;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.moriset.bcephal.domain.BrowserData;
import com.moriset.bcephal.utils.JsonDateDeserializer;
import com.moriset.bcephal.utils.JsonDateSerializer;

/**
 * @author Joseph Wambo
 *
 */
public class BookingBrowserData extends BrowserData {

	@JsonSerialize(using = JsonDateSerializer.class)
	@JsonDeserialize(using = JsonDateDeserializer.class)
	public Date entryDate;
	
	@JsonSerialize(using = JsonDateSerializer.class)
	@JsonDeserialize(using = JsonDateDeserializer.class)
	public Date bookingDate1;
	
	@JsonSerialize(using = JsonDateSerializer.class)
	@JsonDeserialize(using = JsonDateDeserializer.class)
	public Date bookingDate2;
	
	public BigDecimal creditAmount;
	
	public BigDecimal debitAmount;
	
	public String username;
	
	public boolean manual;
	
	public BookingBrowserData() {
		this.creditAmount = BigDecimal.ZERO;
		this.debitAmount = BigDecimal.ZERO;
	}
		
	public BookingBrowserData(Booking booking) {
		super(booking.getId(), booking.getBookingId());	
		this.entryDate = booking.getEntryDate();
		this.bookingDate1 = booking.getBookingDate1();
		this.bookingDate2 = booking.getBookingDate2();
		this.creditAmount = booking.getCreditAmount();
		this.debitAmount = booking.getDebitAmount();
		this.username = booking.getUsername();
		this.manual = booking.isManual();
	}
	
	
}

File: client/bcephal-client-model/Accounting/BookingBrowserData.cs
using Newtonsoft.Json;
using System;

namespace Bcephal.Models.Base.Accounting
{
    public class BookingBrowserData : BrowserData
    {

        public string entryDate { get; set; }
        [JsonIgnore]
        public DateTime? EntryDate
        {
            get
            {
                try
                {
                    return DateUtils.Parse(entryDate);
                }
                catch (Exception)
                {
                    return null;
                }
            }
            set
            {
                try
                {
                    entryDate = DateUtils.Format(value);
                }
                catch (Exception)
                {
                    entryDate = null;
                }
            }
        }

        public string bookingDate1 { get; set; }
        [JsonIgnore]
        public DateTime? BookingDate1
        {
            get
            {
                try
                {
                    return DateUtils.Parse(bookingDate1);
                }
                catch (Exception)
                {
                    return null;
                }
            }
            set
            {
                try
                {
                    bookingDate1 = DateUtils.Format(value);
                }
                catch (Exception)
                {
                    bookingDate1 = null;
                }
            }
        }

        public string bookingDate2 { get; set; }
        [JsonIgnore]
        public DateTime? BookingDate2
        {
            get
            {
                try
                {
                    return DateUtils.Parse(bookingDate2);
                }
                catch (Exception)
                {
                    return null;
                }
            }
            set
            {
                try
                {
                    bookingDate2 = DateUtils.Format(value);
                }
                catch (Exception)
                {
                    bookingDate2 = null;
                }
            }
        }

        [JsonIgnore]
        public decimal BalanceAmount
        {
            get
            {
                return Math.Abs(creditAmount - debitAmount);
            }
        }

        [JsonIgnore]
        public string Sign
        {
            get
            {
                return creditAmount - debitAmount < 0 ? "D" : "C";
            }
        }

        public decimal creditAmount { get; set; }
        
        public decimal debitAmount { get; set; }
        
        public string username { get; set; }

        public bool manual { get; set; }

    }
}
